core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceMultipartTest.java

```diff
@@ -89,6 +90,31 @@ void testUploadFile(String path) throws Exception {
                            "\"param1\":\"armeria\"}");
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = "/uploadWithUnintendedFileParam")
+    void testUploadFileWithUnexpectedParameters(String path) throws Exception {
+        final Multipart multipart = Multipart.of(
+                BodyPart.of(ContentDisposition.of("form-data", "file1", "foo.txt"), "foo"),
+                BodyPart.of(ContentDisposition.of("form-data", "path1", "bar.txt"), "bar"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile1", "qux.txt"), "qux"),
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile2", "quz.txt"),
+                            MediaType.PLAIN_TEXT, "quz"),
+                // Unexpected parameter: multipartFile3
+                BodyPart.of(ContentDisposition.of("form-data", "multipartFile3", "qux3.txt"), "qux3"),
+                BodyPart.of(ContentDisposition.of("form-data", "param1"), "armeria")
+
+        );
+        final AggregatedHttpResponse response =
+                server.blockingWebClient().execute(multipart.toHttpRequest(path));
+        assertEquals(HttpStatus.OK, response.status());
+        assertThatJson(response.contentUtf8())
```

Comment:
In addition to checking the response, can we also check that the file has not been physically written?

e.g.

Setting the strategy to never remove automatically:

```java
    @TempDir
    static java.nio.file.Path sharedTempDir;

    @RegisterExtension
    static final ServerExtension server = new ServerExtension() {
        @Override
        protected void configure(ServerBuilder sb) throws Exception {
            sb.annotatedService("/", new MyAnnotatedService());
            sb.decorator(LoggingService.newDecorator());
            sb.multipartRemovalStrategy(MultipartRemovalStrategy.NEVER);
            sb.multipartUploadsLocation(sharedTempDir);
```

and then, check if any of the temporary files contains "qux3"
```java
        assertThat(sharedTempDir.resolve("complete").toFile().listFiles())
                .noneSatisfy(file -> assertThat(file).content(StandardCharsets.UTF_8).isEqualTo("qux3"));
```